internal/engine/watchmanager.go

```diff
@@ -70,11 +71,33 @@ func (w *WatchManager) diff(ctx context.Context, st *store.Store) (setup []Watch
 	return setup, teardown
 }
 
+func (w *WatchManager) maybeSetupTFWatch(st *store.Store) error {
+	state := st.RLockState()
+	defer st.RUnlockState()
+	if w.tiltfileWatch != nil {
+		return nil
+	}
+
+	watcher, err := w.fsWatcherMaker()
```

Comment:
it seems weird to set up the watcher while you have an RLock on the state?

maybe the diff() method could just return something to indicate that this needs to be setup?

Reply:
Yeah I thought about that but it felt gross. Maybe I can refactor that function a bit. I agree overall.